### PR TITLE
[RATOM-72] Add import stages and initial tests

### PR DESCRIPTION
**Pull request by copelco**
* Implement import stage for each import_status value: Created, Importing, Complete, Failed.
* Add core.models.File.errors JSON field to track and report file-level errors
* Split import components into submodules so it's more manageable
* Validate each row using a Django form, which is more familiar
* Add pytest and tests various aspects of import process
* Add tqdm for progress bar

### CODE REVIEW

**Review by JeremyGibson — APPROVED** [review]
Works and looks great. Nice!

File: README.rst [hunk 1]
```diff
@@ -106,6 +106,18 @@ Development
...
 
 
+Tests
+----------
+
+To run the projet unit tests, use:
+
+```
+make test
+```
```
Comment (JeremyGibson) [inline README.rst:116]: Should we add the note about how to run the enron_dataset_test?

File: core/admin.py [hunk 2]
```diff
@@ -23,3 +22,43 @@ class MessageAdmin(admin.ModelAdmin):
     date_hierarchy = "sent_date"
```
Comment (JeremyGibson) [inline core/admin.py:22]: nice, I hadn't seen this before.

File: README.rst (hunk 1 above)
Comment (copelco) [reply README.rst:116]: Yes! Added

**Review by JeremyGibson — APPROVED** [review]

**Pull request merged**